internal/tool/command.go

```diff
@@ -0,0 +1,160 @@
+package tool
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"os"
+	"os/exec"
+	"strings"
+
+	codacy "github.com/codacy/codacy-engine-golang-seed/v6"
+	docgen "github.com/codacy/codacy-semgrep/internal/docgen"
+	"github.com/samber/lo"
+)
+
+type SemgrepOutput struct {
+	Results []SemgrepResult `json:"results"`
+	Errors  []SemgrepError  `json:"errors"`
+}
+
+type SemgrepResult struct {
+	CheckID       string          `json:"check_id"`
+	Path          string          `json:"path"`
+	StartLocation SemgrepLocation `json:"start"`
+	EndLocation   SemgrepLocation `json:"end"`
+	Extra         SemgrepExtra    `json:"extra"`
+}
+
+type SemgrepLocation struct {
+	Line int `json:"line"`
+}
+
+type SemgrepExtra struct {
+	Message     string `json:"message"`
+	RenderedFix string `json:"rendered_fix,omitempty"`
+}
+
+type SemgrepError struct {
+	Message  string               `json:"message"`
+	Location SemgrepErrorLocation `json:"location"`
+}
+
+type SemgrepErrorLocation struct {
+	Path string `json:"path"`
+}
+
+func executeCommandForFiles(configFile *os.File, toolExecution codacy.ToolExecution, patternDescriptions *[]codacy.PatternDescription, language string, files []string) ([]codacy.Result, error) {
+	semgrepCmd := createCommand(configFile, toolExecution.SourceDir, language, files)
+
+	semgrepOutput, semgrepError, err := runCommand(semgrepCmd)
+	if err != nil {
+		return nil, errors.New("Error running semgrep: " + semgrepError + "\n" + err.Error())
+	}
+
+	output, err := parseCommandOutput(toolExecution.ToolDefinition, patternDescriptions, semgrepOutput)
+	if err != nil {
+		return nil, err
+	}
+	return output, nil
+}
+
+func createCommand(configFile *os.File, sourceDir, language string, files []string) *exec.Cmd {
+	params := createCommandParameters(language, configFile, files)
+	cmd := exec.Command("semgrep", params...)
+	cmd.Dir = sourceDir
+
+	return cmd
+}
+
+func createCommandParameters(language string, configFile *os.File, filesToAnalyse []string) []string {
+	// adding -json parameters
+	cmdParams := []string{
+		"-json", "-json_nodots",
+	}
+	// adding -lang parameter
+	cmdParams = append(
+		cmdParams,
+		"-lang", language,
+	)
+	// adding -rules parameter
+	cmdParams = append(
+		cmdParams,
+		"-rules", configFile.Name(),
+	)
+	// adding files to analyse
+	cmdParams = append(
+		cmdParams,
+		filesToAnalyse...,
+	)
```

Comment:
```suggestion
	cmdParams := []string{
		// adding -json parameters
		"-json", "-json_nodots",
	    // adding -lang parameter
		"-lang", language,
	    // adding -rules parameter
		"-rules", configFile.Name(),
	    // adding files to analyse
		filesToAnalyse...,
	}
```

I'm not sure if the last part is actually possible though, you might need to append in that case.

Also, I would move this to the `createCommand` function, I don't think it's necessary to further split the logic.

Reply:
I changed it, but im going to keep this function here to separate the concerns: this is just parameters (which can change), the other is about creating a command to execute. I think they are related but are different things.